src/decoder/ifd.rs

```diff
@@ -332,171 +332,191 @@ impl Entry {
+                        v.push(Short(r.read_u16()?));
+                    }
+                    return Ok(List(v));
                 }
-                (Type::LONG, 2) => {
+                Type::SSHORT => {
                     let mut r = self.r(bo);
-                    Some(List(vec![Unsigned(r.read_u32()?), Unsigned(r.read_u32()?)]))
+                    let mut v = Vec::new();
+                    for _ in 0..self.count {
+                        v.push(Signed(i32::from(r.read_i16()?)));
+                    }
+                    return Ok(List(v));
                 }
-                (Type::SLONG, 2) => {
+                Type::LONG => {
                     let mut r = self.r(bo);
-                    Some(List(vec![Signed(r.read_i32()?), Signed(r.read_i32()?)]))
+                    let mut v = Vec::new();
+                    for _ in 0..self.count {
+                        v.push(Unsigned(r.read_u32()?));
+                    }
+                    return Ok(List(v));
                 }
-                (Type::RATIONAL, 1) => {
+                Type::SLONG => {
                     let mut r = self.r(bo);
-                    let numerator = r.read_u32()?;
-                    let denominator = r.read_u32()?;
-                    Some(Rational(numerator, denominator))
+                    let mut v = Vec::new();
+                    for _ in 0..self.count {
+                        v.push(Signed(r.read_i32()?));
+                    }
+                    return Ok(List(v));
                 }
-                (Type::SRATIONAL, 1) => {
+                Type::FLOAT => {
                     let mut r = self.r(bo);
-                    let numerator = r.read_i32()?;
-                    let denominator = r.read_i32()?;
-                    Some(SRational(numerator, denominator))
+                    let mut v = Vec::new();
+                    for _ in 0..self.count {
+                        v.push(Float(r.read_f32()?));
+                    }
+                    return Ok(List(v));
                 }
-                _ => None,
+                Type::LONG8 | Type::RATIONAL | Type::SRATIONAL | Type::DOUBLE => unreachable!(),
+                Type::__NonExhaustive => unreachable!(),
             }
-        } else {
-            None
-        };
+        }
 
-        if let Some(v) = bigtiff_value {
-            Ok(v)
-        } else {
-            match (self.type_, self.count) {
-                // TODO check if this could give wrong results
-                // at a different endianess of file/computer.
-                (Type::BYTE, 1) => Ok(Unsigned(u32::from(self.offset[0]))),
-                (Type::BYTE, 2) => offset_to_bytes(2, self),
-                (Type::BYTE, 3) => offset_to_bytes(3, self),
-                (Type::BYTE, 4) => offset_to_bytes(4, self),
-                (Type::BYTE, n) => self.decode_offset(n, bo, limits, decoder, |decoder| {
-                    Ok(UnsignedBig(u64::from(decoder.read_byte()?)))
```

Comment:
It isn't clear to me why this returns UnsignedBig instead of Byte, but we should probably make a decision about whether to retain that going forward.

Reply:
Possibly because `Byte` is newer. I'm not 100% sure about that though.

Reply:
Let's just keep the case here for compatibility. I suppose we should rework the `Value`-system anyways. The allocation can not be re-used and not be done upfront because every different type requires a different `Vec<_>` allocation, instead of a generically aligned byte buffer.